### DISCUSSION TITLE
Github classroom bug: Edit assignment makes starter code repository setting disappear and student will get blank starter code.

### DISCUSSION THREAD

**Discussion by piti118** [community/community#132094] [GitHub Education]
### Hi everyone,

Step to reproduce (14 July)
1) Create an assignment.
  - Here are some may be relevant setting
  - Private template repo.
  - Github preset test. (all custom code (I did go test go-exam/q1 etc.)
2) Student A accepts the assignment.
  - Assignment A is forked correctly
3) Teacher edit assignment in particular I added more tests. (Note that the repo setting still exists here but is not editable. So I cannot put it back manually as well.
4) Click Update Assignment
5) Once you are on the overview page. If you click edit again you will find that the starter code repo setting is now blank. It is also not editable.
6) If student B tries to accept the assignment, one will get a blank assignment starter code.

I tried it at least 3 assignement. Did I miss something why should adding a test make the start code setting disappear? This makes the product essentially unusable at the current stage.

**Comment by brunomateus** [comment]
Same here.
I updated some assignments and all of them presented the same problem.

**Comment by mastarija** [comment]
Definitely an issue. It seems like the edit page erases the starter code repository field, or rather, the back-end erases it once it processes the submitted form making the assignment unusable. This makes it impossible to e.g. update the assignment deadline.

**Comment by mr-hare-cca** [comment]
This is making it really hard to use. One little tiny change makes the assignment unusable... can change the date, name, add tests...

I was super excited to use this, but going have to ditch it if this continues.

How can we escalate this.

**Comment by mh-skjelvareid** [comment]
I have encountered the same exact problem. Also, the testing information (contained in .github/workflows/classroom.yml) is not distributed to the students either.

**Reply by mr-hare-cca** [reply]
I just figured that out too, I was able to copy and paste it into the template repo from the code classroom generates for you when you add tests.

**Reply by mh-skjelvareid** [reply]
Nice, that's kind of a quick fix - but if the link to the template repo in the assignment disappears when it is is edited, so the students don't get the starter code, then they won't get the manually added .github folder either...?

**Reply by mr-hare-cca** [reply]
I also realized it creates a new repo in your organization which is linked at the top of classroom that you can edit. I have them use GitHub Desktop, so I think that adds the .github folder… or maybe I had it in my repo to start. Either way, it’s annoying and not working as intended. Upside, I have learned a lot more about Git and actions.

**Reply by srush** [reply]
So what is the workaround? If you add the .github/ to the start code repo, it will run student tests? Can you post the workflow code? 

Does it still create the fake pull request?

**Comment by Sayali19-cell** [comment]
Visit the GitHub Status page to see if there are any ongoing issues with GitHub services that might be affecting your ability to update assignments.
This issue appears to be a bug related to how GitHub Classroom handles updates to assignments after they’ve been created. By following the troubleshooting steps outlined above, you can attempt to resolve the issue or at least work around it while waiting for a permanent fix from GitHub. If the issue is significant and blocking your work, reaching out to GitHub Support would be a necessary step to get personalized assistance.

**Comment by mr-hare-cca** [comment]
The feedback/pull request functionality doesn't seem to be working either.

**Comment by duckkyzTeacher** [comment]
Any update on this? Its causing huge headaches for me since I am setting up my class this year. Any modification (due dates, tests) requires me to make a new assignment.

**Reply by mr-hare-cca** [reply]
As per the message above, I checked the status page and there doesn’t seem to be any problems… the next step was “reaching out to GitHub Support would be a necessary step to get personalized assistance.” I had briefly tried to do this, but it wouldn’t let me submit it and told me to post it in community boards. 🤷🏼‍♂️

**Reply by duckkyzTeacher** [reply]
Ridiculous, how do they expect teacher to use the platform if it doesnt work and they wont take feedback on it?

Thanks for trying though.

**Comment by mh-skjelvareid** [comment]
This problem has also been reported as an issue in the repo github/gh-classroom :
https://github.com/github/gh-classroom/issues/78

**Comment by DrErickson** [comment]
I ran into this problem when I updated the assignment due date.   The template disappeared and I am unable to add the template back using "Edit assignment"

![image](https://github.com/user-attachments/assets/4e0edfa6-eae4-4900-aeda-4772abd67393)

The "Find a GitHub repository" field is disabled if I try to add back the template.

I hope this issue is resolved soon.

**Comment by brunomateus** [comment]
Nothing has changed? I'm afraid to change the deadline of my assignment and run into the same problem again
